### DISCUSSION TITLE
Redeploy via API after changing env variables?

### DISCUSSION THREAD

**Discussion by aarondfrancis** [vercel/community#252] [Help] [answered]
Hey there,

I understand from the docs that after an env var has been changed, the project will need to be redeployed to utilize it.

Is there a way to trigger a redeploy of a _non git-backed_ project via API? I could read the previous deployment via API and then deploy the same set of files, but I was wondering if that's built in at all?

Thanks!

**Comment by samcx** [comment, marked as answer]
Unfortunately, there is no built-in way trigger a redeploy of a _non git-backed_ project via the API.

In the dashboard, a re-deploy just refetches that same source at that commit, which is why it is not possible to redeploy a _non git-backed_ Deployment from the dashboard.

**Reply by aarondfrancis** [reply]
Cool, no worries. I figured!
